### PR TITLE
MvxSetupSingleton optimizations / Fix SplashScreen initialization on Android

### PR DESCRIPTION
**Pull request by nmilcoff**
### :sparkles: What kind of change does this PR introduce? (Bug fix, feature, docs update...)
Bug fix

### :arrow_heading_down: What is the current behavior?
1. Starting from SplashScreen doesn't work on Android
2. Setup is called multiple times during Android startup

### :new: What is the new behavior (if this is a feature change)?
Above is fixed, also made some optimizations and added some reference comments to MvxSetupSingleton.

### :boom: Does this PR introduce a breaking change?
Yes. TriggerFirstNavigate was removed.

### :bug: Recommendations for testing
Test the Playground apps

### :memo: Links to relevant issues/docs
PR: #2713 

### :thinking: Checklist before submitting

- [X] All projects build
- [X] Follows style guide lines ([code style guide](https://github.com/MvvmCross/MvvmCross#code-style-guidelines))
- [ ] Relevant documentation was updated ([docs style guide](https://www.mvvmcross.com/documentation/contribute/mvvmcross-docs-style-guide))
- [X] Rebased onto current develop

### CODE REVIEW

**Comment by nmilcoff** [comment]
@kjeremy 👍 ! It's splitted now

File: Projects/Playground/Playground.Droid/SplashScreen.cs [hunk 1]
```diff
@@ -15,11 +15,16 @@ namespace Playground.Droid
...
         , ScreenOrientation = ScreenOrientation.Portrait)]
-    public class SplashScreen : MvxSplashScreenActivity
+    public class SplashScreen : MvxSplashScreenAppCompatActivity
     {
         public SplashScreen()
             : base(Resource.Layout.SplashScreen)
         {
         }
+
+        protected override void OnResume()
```
Comment (nickrandolph) [inline Projects/Playground/Playground.Droid/SplashScreen.cs]: Does this override need to be here?

File: MvvmCross/Core/MvxSetupSingleton.cs [hunk 2]
```diff
@@ -61,57 +55,60 @@ public virtual void EnsureInitialized()
...
+                    if (_currentMonitor == null)
+                    {
+                        IsInitialisedTaskCompletionSource.Task.GetAwaiter().GetResult();
+                        MvxLog.Instance.Trace("EnsureInitialized has already been called so now waiting for completion");
+                    }
                 }
                 else
                 {
-                    MvxLog.Instance.Trace("EnsureInitialized has already been called so now waiting for completion");
+                    StartSetupInitialization();
```
Comment (nickrandolph) [inline MvvmCross/Core/MvxSetupSingleton.cs]: This beaks the behaviour of EnsureInitialized which should block until setup is completed. Currently this PR breaks more than it fixes.

File: MvvmCross.Android.Support/V7.AppCompat/MvxSplashScreenAppCompatActivity.cs [hunk 3]
```diff
@@ -86,5 +85,10 @@ protected virtual void TriggerFirstNavigate()
             if (!startup.IsStarted)
                 startup.Start();
         }
+
+        // do nothing on this override, as initial navigation is managed by TriggerFirstNavigate
```
Comment (martijn00) [inline MvvmCross.Android.Support/V7.AppCompat/MvxSplashScreenAppCompatActivity.cs]: Should we remove TriggerFirstNavigate?

File: MvvmCross/Core/MvxSetupSingleton.cs (hunk 2 above)
Comment (nmilcoff) [reply MvvmCross/Core/MvxSetupSingleton.cs]: Sounds reasonable. We need to test PRs like #2713 or this one more before merging...

**Comment by nmilcoff** [comment]
Ok guys, thank you all for your feedback! Honestly I feel like this PR is actually fixing some issues that were introduced just before... We need to be more careful with merging.

You'll notice I removed some calls to EnsureInitialized. This is because all Android Activities are already calling that [here](https://github.com/MvvmCross/MvvmCross/blob/develop/MvvmCross/Platforms/Android/Views/MvxActivityViewExtensions.cs#L138).

**Comment by nickrandolph** [comment]
I think this is probably one of the more complex parts of mvx to get right. Mainly because each platform has slightly different startup behaviour and we’re trying to coerce into a single pattern to make it easy for devs.
I don’t think we’re quite there as there are still some race conditions with the latest edits but we’re getting there. I think it’s be good to have some tests around this, so I will try and get that started when I have some time.

**Comment by kjeremy** [comment]
A big fat high level comment on how all this works might be good.

**Comment by nickrandolph** [comment]
@martijn00 @nmilcoff I've done another pass through this PR and whilst there are some extreme edge cases where a race condition can occur, they're not easily resolvable without reworking the singleton significantly. I propose that this PR is accepted as is, and that we confirm that all platforms (inc Forms) are working well on startup.

**Pull request merged**